src/Evolution/Systems/GrMhd/ValenciaDivClean/PalenzuelaEtAl.cpp

```diff
@@ -150,7 +154,12 @@ std::optional<PrimitiveRecoveryData> PalenzuelaEtAl::apply(
         Scalar<double>(rest_mass_density),
         Scalar<double>(specific_internal_energy)));
   } else if constexpr (ThermodynamicDim == 3) {
-    ERROR("3d EOS not implemented");
+    const double specific_internal_energy = f_of_x.specific_internal_energy(
```

Comment:
clang-tidy rightfully complains here that `specific_internal_energy` already exists above.